blog/2023-05-24-approach-to-devex.md

```diff
@@ -0,0 +1,103 @@
+Instead, they can focus their efforts on crafting application code, while Winglang's compiler handles the cloud mechanics. For example, the automated generation of Identity and Access Management (IAM) policies.
+
+Another key aspect of Winglang's approach to reducing cognitive load is its introduction of [inflights](https://docs.winglang.io/concepts/inflights). They allow developers to write distributed code that looks and feels almost like that of a monolith, with all the cognitive benefits that come with it: code that is easier to follow, test and debug.
+
+Winglang also introduces features such as implicit await for asynchronous calls, further alleviating developers' cognitive burden. 
+
+These cognitive load-reducing attributes seem to align well with the second dimension identified in the academic paper. 
+The challenge is to make sure that the abstractions are not leaky on the one hand and that they don't oversimplify to the degree of diminishing control on the other hand. 
+It's a tough one, and we are not there yet, although I feel like we're on the right path to get there. For example, by supporting import of any Terraform module directly, Wing doesn't restrict developers to using abstractions. There are also many ways to customize and extend Wing, including custom resources and [compiler plugins](https://docs.winglang.io/blog/2023/02/17/plugins) to modify the Terraform output directly.
+
+### Flow State
+Winglang aims to keep developers immersed in their tasks by:
+
+**Providing Autonomy:** Winglang handles cloud mechanics, enabling developers to focus more on the business logic of their applications and less on infrastructure.
+Our goal is to reach a point where they can create entire applications in dev environments without needing much intervention from DevOps.
+On the other hand, Winglang also aims to promote the autonomy of DevOps engineers. [Compiler plugins](https://docs.winglang.io/blog/2023/02/17/plugins) are a way for them to apply non-functional concerns to the application in the form of policies, without having to communicate with the application developers much or to know every resource being used by them.
+ 
+If we succeed in making developers and DevOps engineers more autonomous, we will be able to reduce the number of context switches and hand-offs between them. This should help induce a flow state in both.
+
+**Minimizing Disruptions:** In Winglang both application and infrastructure code are composed in the same way and using the same programming model and tools. 
+This reduces context switching, whether between languages, tools, or infra and application code. 
+If different developers are handling infra and app code, Winglang reduces the number of context switches associated with communication between them of the separation of concerns and increased autonomy for both disciplines described above. 
```

Comment:
```suggestion
If different developers are handling infra and app code, Winglang reduces the number of context switches associated with communication between them because of the separation of concerns and increased autonomy for both disciplines described above. 
```